app/utils/SystemPropertiesHelper.scala

```diff
@@ -36,13 +36,13 @@ trait SystemPropertiesHelper extends Logging {
       case t: Throwable => doesntParse(key, defaultValue, t.getMessage)
     }
 
-  def getStringProperty(key: String, defaultValue: String): String =
+  def getStringProperty(key: String, defaultValue: Option[String]): String =
     try if (sysprop.get(key).isDefined) {
-      sysprop.get(key).getOrElse(defaultValue)
+      sysprop.get(key).getOrElse(defaultValue.get)
```

Comment:
to do a getOrElse with an optional value feels weird :). I think the code was fine before. The test was wrong as the compiler would pick up an issue if the defaultValue was not provided.